spec/requests/chat_channel_controller_spec.rb

```diff
@@ -656,4 +656,49 @@
       expect(channel.reload.status).to eq("open")
     end
   end
+
+  describe "#delete" do
+    fab!(:channel) do
+      Fabricate(
+        :chat_channel,
+        chatable: topic,
+        name: "Ambrose Channel",
+        status: :open
+      )
+    end
+
+    it "returns error if user is not admin" do
+      sign_in(user)
+      delete "/chat/chat_channels/#{channel.id}.json", params: { channel_name_confirmation: "ambrose channel" }
+      expect(response.status).to eq(403)
+    end
+
+    it "returns a 404 if the channel does not exist" do
+      channel.destroy!
+      sign_in(admin)
+      delete "/chat/chat_channels/#{channel.id}.json", params: { channel_name_confirmation: "ambrose channel" }
+      expect(response.status).to eq(404)
+    end
+
+    it "returns a 400 if the channel_name_confirmation does not match the channel name" do
+      sign_in(admin)
+      delete "/chat/chat_channels/#{channel.id}.json", params: { channel_name_confirmation: "some Other channel" }
+      expect(response.status).to eq(400)
+    end
+
+    it "deletes the channel right away and enqueues the background job to delete all its chat messages and related content" do
+      sign_in(admin)
+      delete "/chat/chat_channels/#{channel.id}.json", params: { channel_name_confirmation: "ambrose channel" }
+      expect(response.status).to eq(200)
+      expect(ChatChannel.find_by(id: channel.id)).to eq(nil)
```

Comment:
```suggestion
      expect(channel.reload.trashed?).to eq(true)
```

I think this is a more accurate representation of what we're asserting against. The query assumes the `default_scope` context exists on the model.

Reply:
It always does though, I don't think that's a bad assumption to make if we trust `Trashable`

Reply:
But why introduce an indirection when we can assert for the state of the object directly?

Reply:
There is an additional level of indirection introduced here though compared to asserting for the state of the object directly.

Reply:
Okay understood, will change :+1: